legacy/src/main/java/org/opensearch/sql/legacy/executor/join/NestedLoopsElasticExecutor.java

```diff
@@ -49,6 +48,8 @@ public NestedLoopsElasticExecutor(Client client, NestedLoopsElasticRequestBuilde
 
   @Override
   protected List<SearchHit> innerRun() throws SqlParseException {
+    createPointInTimeHandler(client, nestedLoopsRequest);
+    pit.create();
```

Comment:
Instead of doing this in each subclass, we can do it in parent class's `run()` with try-finally to make sure PIT will be released?

Reply:
addressed this in https://github.com/opensearch-project/sql/pull/2703/commits/617d1dbab3875636e522c522f99e42691d587ae1